backend/tests/clubs/test_views.py

```diff
@@ -1620,6 +1622,95 @@ def test_club_invite_email_resend(self):
         )
         self.assertIn(resp.status_code, [200, 201], resp.content)
 
+    def test_send_invite_with_acceptance_email(self):
+        self.client.login(username=self.user1.username, password="test")
+
+        # Create a club application
+        now = timezone.now()
+        application = ClubApplication.objects.create(
+            name="Send Acceptance Invivites",
+            club=self.club1,
+            application_start_time=now,
+            application_end_time=now + timezone.timedelta(days=1),
+            result_release_time=now + timezone.timedelta(days=2),
+            is_wharton_council=False,
+        )
+        application.save()
+
+        applicants = [self.user1, self.user2, self.user3, self.user4]
+        # 4 = accepted, 2 = rejected, 1 = pending
+        statuses = [4, 4, 2, 1]
+
+        # Create submissions for each applicant
+        i = 0
+        for user in applicants:
+            submission = ApplicationSubmission.objects.create(
+                user=user,
+                application=application,
+                committee=None,
+                status=statuses[i],
+                reason="a reason",
+            )
+            i += 1
+            submission.save()
+
+        # Send acceptance & invitations emails emails
+        resp = self.client.post(
+            reverse(
+                "club-applications-send-emails",
+                kwargs={"club_code": self.club1.code, "pk": application.pk},
+            ),
+            {
+                "allow_resend": True,
+                "dry_run": False,
+                "email_type": {"id": "acceptance", "name": "Acceptance"},
+            },
+            content_type="application/json",
+        )
+        self.assertIn(resp.status_code, [200, 201], resp.content)
+
+        # Check club application was created
+        self.assertTrue(
+            ClubApplication.objects.filter(
+                club=application.club, name="Send Acceptance Invivites"
+            ).exists()
+        )
+
+        i = 0
+        for user in applicants:
+            # check application submissions were created
+            self.assertTrue(
+                ApplicationSubmission.objects.filter(
+                    application=application, user=user, status=statuses[i]
+                ).exists()
+            )
+            i += 1
```

Comment:
It's good to be thorough, but I think testing for ancillary objects (i.e. the club application, application submissions) is out of scope for this test. Ideally, a unit test should test a single feature/component, and do it as efficiently as possible.

Since the goal of this test is to check for the creation of membership invites, these checks are unnecessary. In 99.99% of cases, we can expect that calling `ApplicationSubmission.objects.create()` will work as intended.